### DISCUSSION TITLE
Copying a citation vs copying a bibliography item

### DISCUSSION THREAD

**Discussion by przemyslslaw** [retorquere/zotero-better-bibtex#2475] [General]
Hi, 

Thank you for developing this very useful plug-in. 

I've set it up so that Ctrl+Shift+C copies the Latex citation, e.g., `\cite{Einstein}`, but what I'd like to have instead is use Ctrl+Shift+A for the citation, while Ctrl+Shift+C for the human-readable bibliography item, e.g., `Einstein, Albert (1931). On the cosmological problem of the general theory of relativity. Sonderasugabe aus den Sitzungsb. König. Preuss. Akad.: 235–237`, since I use both in my workflows.

Is this possible to do this? If not, then how else can I quickly copy either of these two? In Mendeley, there is a "Copy as..." dialog accessible under right click, which makes both available, but this feature doesn't exist in Zotero or I don't know how to access it yet.

Your help would be greatly appreciated!

**Comment by retorquere** [comment]
Zotero has chosen to omit the infrastructure for this kind of thing. It'd be hard and fragile to break into that, sorry.

**Reply by przemyslslaw** [reply]
Couldn't BetterBibtex assign two different actions (citation and bibliography item) to the two hotkeys, similarly to how Zotero does it? Ideally, users would be able to pick what action they want for each of the two hotkeys. BetterBibtex already allows this choice via its menu, but currently it doesn't distinguish between the two hotkeys.

**Reply by retorquere** [reply]
The hotkey handling happens deep inside a function in Zotero where I can't easily change Zotero's behavior. The one path (ctrl-A or ctrl-C, I forget which) just has a hard "do nothing here" if you choose a translator instead of a CSL style. There's nothing to assign any behavior to for that shortcut when a translator is selected.

BBT isn't involved in shortcut handling in any way. Zotero has implemented shortcuts, and you can either assign a CSL style to these shortcuts, which gives you both behaviors, or a translator, which gives you exactly one. You get to pick the translator, and if you pick BBT, that gets ran, but BBT does not know whether it is being invoked for a shortcut, a regular export or something else.

If Zotero allowed you to pick a 2nd translator for the other shortcut, it would Just Work, and I would have to do nothing else.

**Reply by przemyslslaw** [reply]
Thank you for this detailed answer! That's a bummer. I'll open a Zotero feature request. Honestly, I may end up continuing to use Mendeley because of this :-/

**Reply by retorquere** [reply]
It'd be a relatively simple change as far as I can tell. There'd need to be a selection list to select the 2nd translator, and a few lines copied to call the same code, just with the 2nd translator when "asCitations" mode is selected.

**Reply by retorquere** [reply]
It looks like Zutilo has the option to run a 2nd translator: https://github.com/wshanks/Zutilo/blob/master/docs/COMMANDS.md (search for **QuickCopy items**). I don't know what shortcut it assigns for it though.

**Reply by przemyslslaw** [reply]
Wonderful -- I'll give it a shot -- many thanks!
